Scripts/BuildEngine.ps1

```diff
@@ -24,151 +24,160 @@
+                 }
+                2022 { 
+                    $CMakeGenerator = "-G `"$VS17`" -A $ARCHITECTURE"
+                 }
+                Default {
+                    throw 'This version of Visual Studio is not supported' 
+                }
+            }
+          
+            $CMakeCacheVariableOverride = $CMakeCacheVariableOverride,'DCMAKE_CONFIGURATION_TYPES=Debug;Release' -join " -"
+         }
+
+         "Linux" { 
+            $CMakeGenerator = "-G `"$CMAKE_GENERATOR_UNIX`""
+            $CMakeCacheVariableOverride = $CMakeCacheVariableOverride,'DSDL_STATIC=ON','DSDL_SHARED=OFF' -join " -"
+
+            #Set Linux build compiler
+            $env:CC= '/usr/bin/gcc-11'
+            $env:CXX= '/usr/bin/g++-11'
+        }
+        "Darwin" { 
+            $CMakeGenerator = "-G `"$CMAKE_GENERATOR_DARWIN`""
+            $CMakeCacheVariableOverride = $CMakeCacheVariableOverride,'DBUILD_FRAMEWORK=ON' -join " -"
+        }
+        Default {
+            throw 'This system is not supported'
+        }
     }
 
-    $CMakeCacheVariableOverride += " -DCMAKE_SYSTEM_NAME=$systemName"
-    $CMakeCacheVariableOverride += " -DCMAKE_BUILD_TYPE=$configuration"
-    $CMakeCacheVariableOverride += " -DBUILD_SANDBOX_PROJECTS=ON"
-    $CMakeCacheVariableOverride += " -DENTT_INCLUDE_HEADERS=ON"
-    
-    # SDL2 options
-    #    
-    if ($systemName -eq "Linux") {
-        $CMakeCacheVariableOverride += " -DSDL_STATIC=ON"
-        $CMakeCacheVariableOverride += " -DSDL_SHARED=OFF"
-    }
-    
-    # Spdlog options
-    #
-    $CMakeCacheVariableOverride += " -DSPDLOG_BUILD_SHARED=OFF"
-    $CMakeCacheVariableOverride += " -DBUILD_STATIC_LIBS=ON"
-    $CMakeCacheVariableOverride += " -DSPDLOG_FMT_EXTERNAL=ON"
-    $CMakeCacheVariableOverride += " -DSPDLOG_FMT_EXTERNAL_HO=OFF"
-    
-    # GLFW options
-    #
-    if ($systemName -ne "Linux") {
-        $CMakeCacheVariableOverride += " -DGLFW_BUILD_DOCS=OFF"
-        $CMakeCacheVariableOverride += " -DGLFW_BUILD_EXAMPLES=OFF"
-        $CMakeCacheVariableOverride += " -DGLFW_INSTALL=OFF"
-    }
+    $CMakeCacheVariableOverride = $CMakeCacheVariableOverride,('DCMAKE_SYSTEM_NAME='+$SystemName),
+    ('DCMAKE_BUILD_TYPE='+$configuration),'DBUILD_SANDBOX_PROJECTS=ON','DENTT_INCLUDE_HEADERS=ON','DSPDLOG_BUILD_SHARED=OFF','DBUILD_STATIC_LIBS=ON','DSPDLOG_FMT_EXTERNAL=ON','DSPDLOG_FMT_EXTERNAL_HO=OFF','DASSIMP_BUILD_TESTS=OFF','DASSIMP_INSTALL=OFF','DASSIMP_BUILD_SAMPLES=OFF','DASSIMP_BUILD_ASSIMP_TOOLS=OFF','DUUID_BUILD_TESTS=OFF','DUUID_USING_CXX20_SPAN=ON' -join " -"
 
-    # ASSIMP options
-    #
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_TESTS=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_INSTALL=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_SAMPLES=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_ASSIMP_TOOLS=OFF"
-    
-    if ($systemName -eq "Darwin") {
-        $CMakeCacheVariableOverride += " -DBUILD_FRAMEWORK=ON"
+    if(!$IsLinux){
```

Comment:
`if (-not $IsLinux)`

Reply:
why this preference ?

Reply:
since we use operator like -eq, -like to make the comparison clear for everyone, it's convenient to continue  the use of same pattern